### PR TITLE
Hvac Improvements

### PR DESCRIPTION
**Pull request by JasperHoffmann**
## HVAC environment improvements

> **Note:** This PR depends on #272  and should be merged after it.

Refactors and extends the HVAC RL environment: updates house parameters to
Freiburg climate, introduces dict observations, adds weekday occupancy gains,
improves dataset/episode length defaults, makes dynamics CasADi SX-friendly,
adds render info for dynamics params, refines default settings and comfort
bounds, updates the parameter interface, and renames the temperature cost
weight to log-space (`log_q_Ti`). Adds HVAC training scripts and tests.

### Commits
- [e1513a1](https://github.com/leap-c/leap-c/pull/273/commits/e1513a1632d127e750d8f7bbf8bc3f65f2f149fb)  feat(hvac): update house parameters to Freiburg
- [0084084](https://github.com/leap-c/leap-c/pull/273/commits/008408488be1ef352a4193df03b9a79f05da07ed) feat(hvac): update extractor with instance normalization
- [dea4cd2](https://github.com/leap-c/leap-c/pull/273/commits/dea4cd22c425a32bdeab98ef8841c610f4fe983a) feat(hvac): made dynamics SX friendly
- [d47cf42](https://github.com/leap-c/leap-c/pull/273/commits/d47cf42b092c1e45d9a90729ed0739c21f758695) feat(hvac): introduce dict obs
- [af1f02a](https://github.com/leap-c/leap-c/pull/273/commits/af1f02ae539e0a351db7dfa4da32b8404060584b) feat(hvac): add during the week gains to env
- [687b315](https://github.com/leap-c/leap-c/pull/273/commits/687b315f2c887d4547fcf059e20ae99e7f1a573c) feat(hvac): small fixes to dataset, min_price and episode length
- [a0aa146](https://github.com/leap-c/leap-c/pull/273/commits/a0aa146bfb31fd153aca9d1469c49f85957cb7e3) feat(hvac): Add dynamics param to render info
- [e06e8f6](https://github.com/leap-c/leap-c/pull/273/commits/e06e8f6d48b5075a0c45ede1b4b2f5830ba14ece) refactor(hvac): updated default settings of env
-  [626604a](https://github.com/leap-c/leap-c/pull/273/commits/626604a4a70dd97d69a5534ade48de607b648042) refactor(hvac): updated comfort temperature bound
-  [a2a26f9](https://github.com/leap-c/leap-c/pull/273/commits/a2a26f9e3992d397e27c5175377d4aec15817eaf) refactor(hvac): different default param interface
- [7e249cb](https://github.com/leap-c/leap-c/pull/273/commits/7e249cb92d76a04790ef53d468bba428c1dd8cd5) feat(hvac): add hvac scripts
- [2557ec2](https://github.com/leap-c/leap-c/pull/273/commits/2557ec27e4d510aca4745d028595ccfd078a1156) refactor(hvac): rename q_Ti to log_q_Ti in cost param

### Files changed
- `leap_c/examples/hvac/` (acados_ocp, env, planner, dynamics, dataset, utils)
- `leap_c/torch/nn/extractor.py`
- `scripts/hvac/run_sac.py`, `scripts/hvac/run_crossq_zop.py`
- `tests/leap_c/examples/test_hvac.py`

### CODE REVIEW

**Comment by JasperHoffmann** [comment]
I think that happened because of cherry-picking. Just updated them correctly.

**Review by dirkpr — APPROVED** [review]
Looks good! The dict observations simplify a lot. I think `Extractor` implementations that are specific to an example should be placed in that folder. Will fix that in a following PR.
